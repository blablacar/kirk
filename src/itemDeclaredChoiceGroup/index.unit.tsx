import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import ItemDeclaredChoice from 'itemDeclaredChoice'

import ItemDeclaredChoiceGroup, { ItemDeclaredChoiceGroupProps } from './index'

describe('ItemDeclaredChoiceGroup', () => {
  const defaultProps: ItemDeclaredChoiceGroupProps = {
    name: 'name',
    value: 2,
    onChange() {},
    status: ItemDeclaredChoice.STATUS.DEFAULT,
    children: [
      <ItemDeclaredChoice label="1" value={1} />,
      <ItemDeclaredChoice label="2" value={2} />,
      <ItemDeclaredChoice label="3" value={3} />,
    ],
  }
  it('Should map its children and render them with specific props', () => {
    const itemDeclaredChoiceGroup = renderer
      .create(<ItemDeclaredChoiceGroup {...defaultProps} />)
      .toJSON()
    expect(itemDeclaredChoiceGroup).toMatchSnapshot()
  })
  it('Should listen to all children ItemDeclaredChoice changes', () => {
    const itemDeclaredChoiceGroup = shallow(<ItemDeclaredChoiceGroup {...defaultProps} />)
    const onChangeListener = itemDeclaredChoiceGroup.instance().onChange
    expect(itemDeclaredChoiceGroup.find({ value: 1 }).prop('onChange')).toEqual(onChangeListener)
    expect(itemDeclaredChoiceGroup.find({ value: 2 }).prop('onChange')).toEqual(onChangeListener)
    expect(itemDeclaredChoiceGroup.find({ value: 3 }).prop('onChange')).toEqual(onChangeListener)
  })
  it('Should change the value in the state when an ItemDeclaredChoice triggers a change', () => {
    const itemDeclaredChoiceGroup = shallow(<ItemDeclaredChoiceGroup {...defaultProps} />)
    itemDeclaredChoiceGroup.instance().onChange({ name: 'name', value: 'foo' })
    expect(itemDeclaredChoiceGroup.state('value')).toBe('foo')
  })
  it('Should call the onChange prop when an ItemDeclaredChoice triggers a change', () => {
    const onChangeMock = jest.fn()
    const itemDeclaredChoiceGroup = shallow(
      <ItemDeclaredChoiceGroup {...defaultProps} onChange={onChangeMock} />,
    )
    itemDeclaredChoiceGroup.instance().onChange({ name: 'name', value: 'foo' })
    expect(onChangeMock).toHaveBeenCalledWith({ name: defaultProps.name, value: 'foo' })
  })
})
