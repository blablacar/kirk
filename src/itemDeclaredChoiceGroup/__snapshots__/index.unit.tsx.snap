// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDeclaredChoiceGroup Should map its children and render them with specific props 1`] = `
Array [
  <label
    className="jsx-1542968643 kirk-item kirk-item-declared-choice"
  >
    <div
      className="jsx-1542968643 kirk-item-leftText"
    >
      <div
        className="kirk-text kirk-text-body kirk-item-body"
      >
        1
      </div>
    </div>
    <div
      className="jsx-1542968643 kirk-item-rightAddon"
    >
      <input
        checked={false}
        disabled={false}
        name="name"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <svg
        aria-hidden={true}
        className="jsx-2358291114 kirk-icon thin"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="jsx-2358291114"
          cx="33"
          cy="33"
          fill="none"
          r="30"
          stroke="#00AFF5"
        />
      </svg>
    </div>
  </label>,
  <label
    className="jsx-1542968643 kirk-item kirk-item-declared-choice"
  >
    <div
      className="jsx-1542968643 kirk-item-leftText"
    >
      <div
        className="kirk-text kirk-text-body kirk-item-body"
      >
        2
      </div>
    </div>
    <div
      className="jsx-1542968643 kirk-item-rightAddon"
    >
      <input
        checked={true}
        disabled={false}
        name="name"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <svg
        aria-hidden={true}
        className="jsx-1043659059 jsx-59167708 kirk-icon"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <path
          className="jsx-1043659059 jsx-59167708"
          d="M6.5 12.5l4 4 8-8"
          fill="none"
          stroke="#FFF"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeMiterlimit="10"
          strokeWidth="1"
        />
      </svg>
    </div>
  </label>,
  <label
    className="jsx-1542968643 kirk-item kirk-item-declared-choice"
  >
    <div
      className="jsx-1542968643 kirk-item-leftText"
    >
      <div
        className="kirk-text kirk-text-body kirk-item-body"
      >
        3
      </div>
    </div>
    <div
      className="jsx-1542968643 kirk-item-rightAddon"
    >
      <input
        checked={false}
        disabled={false}
        name="name"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <svg
        aria-hidden={true}
        className="jsx-2358291114 kirk-icon thin"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="jsx-2358291114"
          cx="33"
          cy="33"
          fill="none"
          r="30"
          stroke="#00AFF5"
        />
      </svg>
    </div>
  </label>,
]
`;
